agents/diamclient.go

```diff
@@ -74,7 +77,10 @@ func NewDiameterClient(addr, originHost, originRealm string, vendorId int, produ
 		},
 	}
 	if len(dictsDir) != 0 {
-		if err := loadDictionaries(dictsDir, "DiameterClient"); err != nil {
+		var err error
```

Comment:
please remove both errors from here and define err in the constructor signature.